spec/statement.dd

```diff
@@ -1103,9 +1103,17 @@ $(GNAME StatementNoCaseNoDefault):
     $(GLINK ScopeBlockStatement)
 )
 
-        $(P $(EXPRESSION) is evaluated. The result type T must be
-        of integral type or $(CODE char[]), $(CODE wchar[]) or $(CODE dchar[]).
-        The result is
+        $(P $(I Expression) is evaluated.
+        If the type of the $(I Expression) is an enum,
+        it is repeatedly converted
+        to its $(GLINK2 enum, EnumBaseType) until it isn't. Then,
+        if the type is of integral type,
+        the $(I Expression) undergoes $(DDSUBLINK spec/type, integer-promotions, Integer Promotions).
+        Then, the type of the $(I Expression) must be either of integral type or
```

Comment:
"the type of. [...] must be [...] of integral type" again does not work, so perhaps: "Then, the resulting type must be either integral or ..."